### ISSUE TITLE
Accept readonly arrays in schema options

### ISSUE THREAD

**Issue by dev054** [cyrilletuzi/angular-async-local-storage#470]
# Checks before posting an issue

- [x] I have read all the library's documentation (README, CHANGELOG, MIGRATION and /docs)
- [x] I understand I am writing to a *human being*, doing this open source project on his *free* time
- [x] I will be involved in my own issue, by answering questions and informing if the issue is resolved

# Configuration

- `@ngx-pwa/local-storage` version: 10.x 
- Angular version (`ng version`): 10.x

# Description of the issue
Currently the options `enum`, `items` and `required` only support mutable arrays. 

# How to reproduce the issue
Example of `enum`:

```
const themes = ['dark', 'light'] as const;
this.storageMap.get<Theme>('theme', { enum: themes }) 
// The type 'readonly ["dark", "light"]' is 'readonly' and cannot be assigned to the mutable type 'string[]'.
```

**Comment by cyrilletuzi** [comment]
Note for myself: check upon TypeScript 4.1 due to microsoft/TypeScript#39258.

**Comment by cyrilletuzi** [comment]
Released in v11.0.0-1.
